### ISSUE TITLE
proposal: Add type position to guru describe output

### ISSUE THREAD

**Issue by hummerd** [golang/go#27308]
It would be nice to have command like 'Go to type definition' in code editors. To help implement this we need to know where type of current (under cursor) identifier is defined.

So proposal is to add "typepos" to json output of describe identifier.
`
{
	"desc": "identifier",
	"pos": "testdata/src/describe-json/main.go:16:8",
	"detail": "value",
	"value": {
		"type": "I",
		"typepos": "testdata/src/describe-json/main.go:21:6",
		"objpos": "testdata/src/describe-json/main.go:12:6"
	}
}
`

**Comment by ianlancetaylor** [comment]
CC @alandonovan

**Comment by myitcv** [comment]
Also cc @ianthehat

**Comment by rsc** [comment]
Ping @alandonovan and @ianthehat. If you decide this is OK or decide what you want instead, it sounds like @hummerd has an implementation.

**Comment by rsc** [comment]
Ping @alandonovan and @ianthehat.

**Comment by alandonovan** [comment]
Yes, it seems like a reasonable feature. If you send me the CL I will review it. Thank you.

BTW, longer term, the Go team at Google will maintain a canonical LSP (language server protocol) implementation, which will subsume most of the features of Guru. In hindsight some of its features such as pointer analysis-based callgraph alias queries do not carry their weight (performance, implementation complexity) and should be dropped or replaced by simpler approximations. I expect that the feature described in this issue is part of the core LSP feature set.

**Comment by shankaryoga21** [comment]
I am not understand, I go google or Microsoft ??
